properties.md

```diff
@@ -0,0 +1,140 @@
+
+Much like HTML allows identifiers for elements (`id`, `class`, etc) that are then independently interpreted by other consumers (javascript, css, etc), or XML with its `attribute=value` pairs, `properties` get their meaning neither from the bundle schema nor the catalog version they happen to be included in.
+
+A similar approach within operator-framework has been taken with olm's spec and status `descriptors`, which annotate fields of spec/status with opaque tokens to be interpreted by consumers. The primary consumer of descriptors is the OpenShift console to display custom UI for new APIs defined by CRDs, but the mechanism is designed to allow other consumers to interpret descriptors as desired: a CLI tool may use some of the information for generating prompts, or use more of it to provide a full console UI.
+
+More examples of the pattern may be found [here](https://en.wikipedia.org/wiki/Attribute%E2%80%93value_pair).
+
+### The resolver consumer
+
+One major consumer of bundle properties is OLM, which reads them to determine the surface that is exposed to the dependency resolver.
+
+The resolver and the registry apis already deal with certain aspects of a bundle via opaque `properties` via the `ListBundles` grpc API:
+
+```proto
+message Property{
+  string type = 1;
+  string value = 2;
+}
+
+message Bundle{
+    ... 
+	repeated Property properties = 12;
+    ...
+}
+```
+
+Many of the other bundle fields are replicated as properties. Here are the properties for an example bundle, rendered as psuedo-`json`:
+
+```json5
+{
+    "type":  "olm.package",
+    "value": `{"packageName":"etcd","version":"0.9.2"}`,
+},
+{
+    "type":  "olm.gvk",
+    "value": `{"group":"etcd.database.coreos.com","kind":"EtcdCluster","version":"v1beta2"}`,
+},
+{
+    "type":  "olm.gvk",
+    "value": `{"group":"etcd.database.coreos.com","kind":"EtcdRestore","version":"v1beta2"}`,
+},
+{
+    "type":  "olm.gvk",
+    "value": `{"group":"etcd.database.coreos.com","kind":"EtcdBackup","version":"v1beta2"}`,
+},
+{
+    "type":  "olm.label",
+    "value": `{"label":"testlabel"}`,
+}
+```
+
+Properties intentionally closely align with the format for [`dependencies.yaml`](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#bundle-dependencies).
+
+Today, all properties are derived from other fields in a bundle. There is no way to write a new property `type` for a consumer (say, a newer version of the resolver, or an external tool that processes sets of bundles).
+
+### Other consumers
+
+Catalog and Bundles are consumed in many other contexts than the resolver. The package-server component presents catalog content for the purpose of selecting packages and channels, for example. 
```

Comment:
It is not clear for me. I understand that bundles are only useful for OLM. I mean, it is not useful for any other project. Are you trying to specify as a consumer here the OLM pieces/projects and/or any tools/components from OF org that works with bundles as a helper for OLM?

Reply:
We are, I believe, attempting to separate concerns for the thing that is currently called "OLM" that is actually several different controllers. I think the goal would be to make components that are independently useful. For example, the resolver component that determines if the set of cluster and user constraints matches with a set of bundles that are available to install seems like it could very useful outside the scope of an on cluster catalog-operator. A real use case would be a kubectl extension that could tell a user if an operator has all of its constraints satisfied by a cluster so that you could install that operator, or even tell you what set of dependencies would be installed if you attempted to install that operator.

Yes, in general the obvious use cases here are for OLM related on cluster components, but providing this level of abstraction allows us to more easily break up those components into independently useful tools. It also enables a future where OLM can, for example, resolve arbitrary helm charts or hook into first class visualization tools without needing to understand internal APIs that are subject to change.